data-node/src/main/java/org/graylog/datanode/opensearch/cli/OpensearchCommandLineProcess.java

```diff
@@ -59,35 +60,30 @@ public class OpensearchCommandLineProcess implements Closeable {
     /**
      * as long as OpenSearch is not supported on macOS, we have to fix the jdk path if we want to
      * start the DataNode inside IntelliJ.
-     *
-     * @param config
      */
-    private void fixJdkOnMac(final OpensearchConfiguration config) {
-        final var isMacOS = OS.isFamilyMac();
-        final var jdk = config.getOpensearchDistribution().directory().resolve("jdk.app");
-        final var jdkNotLinked = !Files.exists(jdk);
-        if (isMacOS && jdkNotLinked) {
-            // Link System jdk into startup folder, get path:
-            final ProcessBuilder builder = new ProcessBuilder("/usr/libexec/java_home");
-            builder.redirectErrorStream(true);
-            try {
-                final Process process = builder.start();
-                final BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream(), Charset.defaultCharset()));
-                var line = reader.readLine();
-                if (line != null && Files.exists(Path.of(line))) {
-                    final var target = Path.of(line);
-                    final var src = Files.createDirectories(jdk.resolve("Contents"));
-                    Files.createSymbolicLink(src.resolve("Home"), target);
-                } else {
-                    LOG.error("Output of '/usr/libexec/java_home' is not the jdk: {}", line);
-                }
-                // cleanup
-                process.destroy();
-                reader.close();
-            } catch (IOException e) {
-                LOG.error("Could not link jdk.app on macOS: {}", e.getMessage(), e);
+    private Environment fixJdkOnMac(Environment environment) {
+        if (!OS.isFamilyMac()) {
+            return environment;
+        }
+        // set system jdk as OS jdk
+        final ProcessBuilder builder = new ProcessBuilder("/usr/libexec/java_home");
+        builder.redirectErrorStream(true);
+        try {
+            final Process process = builder.start();
+            final BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream(), Charset.defaultCharset()));
+            var line = reader.readLine();
+            if (line != null && Files.exists(Path.of(line))) {
+                return environment.withOpensearchJavaHome(Path.of(line));
```

Comment:
Should we check here that the returned line is indeed a readable dir, before we use it?

Reply:
I wouldn't go that far, especially since it is for development only. 

`java_home` is usually used like this, with no additional checks. 
`` export JAVA_HOME=`/usr/libexec/java_home` ``

From the manpage:
```
DESCRIPTION
       The java_home command returns a path suitable for setting the JAVA_HOME environment variable. 
```

If the path exists, I think we should assume it is readable.

Reply:
:+1: